plinth/modules/firewall/templates/firewall.html

```diff
@@ -40,48 +40,54 @@
 
     <div class="row">
       <div class="col-lg-8">
-        <table class='table table-autowidth table-striped'>
+        <table class='table table-autowidth'>
           <thead>
             <th>{% trans "Service/Port" %}</th>
             <th>{% trans "Status" %}</th>
           </thead>
 
           <tbody>
             {% for service in services|dictsort:"name" %}
-              {% if service.ports %}
-                <tr>
-                  <td><strong>{{ service.name }}</strong></td>
-                  <td>
-                    {% if service.is_enabled %}
-                      <span class='label label-success'>
-                        {% trans "Enabled" %}</span>
-                    {% else %}
-                      <span class='label label-warning'>
-                        {% trans "Disabled" %}</span>
-                    {% endif %}
-                  </td>
-                </tr>
-
-                {% for port in service.ports %}
+              {% if service.is_enabled %}
+                {% if service.ports %}
                   <tr>
-                    <td class='cell-indented'><em>{{ port }}</em></td>
-                    <td>
-                      {% if port in internal_enabled_services and port in external_enabled_services %}
+                    <td><strong>{{ service.name }}</strong></td>
+                    <td >
+                      {% if service.is_enabled %}
                         <span class='label label-success'>
-                          {% trans "Permitted" %}</span>
-                      {% elif port in internal_enabled_services %}
-                        <span class='label label-warning'>
-                          {% trans "Permitted (internal only)" %}</span>
-                      {% elif port in external_enabled_services %}
-                        <span class='label label-warning'>
-                          {% trans "Permitted (external only)" %}</span>
+                          {% trans "Enabled" %}</span>
                       {% else %}
-                        <span class='label label-danger'>
-                          {% trans "Blocked" %}</span>
+                        <span class='label label-warning'>
+                          {% trans "Disabled" %}</span>
                       {% endif %}
                     </td>
+                    <td>
+                      <button data-toggle="collapse" data-target=".{{service.service_id}}" class="btn btn-primary pull-right">View Ports</button>
```

Comment:
I wonder if we could streamline this a little. Could the "View Ports" button be made into a [+] icon to the left of the service name?